roles/cifmw_cephadm/tasks/configure_object.yml

```diff
@@ -38,20 +38,50 @@
     output_dir: "/home/zuul/ci-framework-data/artifacts"
     script: |-
       oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack service create --name swift --description 'OpenStack Object Storage' object-store
-      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user create --project service --password {{ cifmw_ceph_rgw_keystone_psw }}  swift
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user create --project {{ project_service_uuid.stdout }} --password {{ cifmw_ceph_rgw_keystone_psw }}  swift
       oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role create swiftoperator
       oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role create ResellerAdmin
-      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user swift --project service member
-      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user swift --project service admin
-      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack endpoint create --region regionOne object-store public http://{{ cifmw_cephadm_vip }}:8080/swift/v1/AUTH_%\(tenant_id\)s
-      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack endpoint create --region regionOne object-store internal http://{{ cifmw_cephadm_vip }}:8080/swift/v1/AUTH_%\(tenant_id\)s
+  delegate_to: localhost
+  when:
+    - cifmw_openshift_kubeconfig is defined
+    - cifmw_ceph_daemons_layout.rgw_enabled | default(true) | bool
+    - not swift_in_ctlplane.stdout | bool
+    - swift_endpoints_count.stdout == "0"
+
+- name: Capture required UUIDs of swift user/service
+  delegate_to: localhost
+  block:
+    - name: Get uuid for user 'swift'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user show swift -c id -f value"
+      register: user_swift_uuid
+
+    - name: Get uuid for service 'swift'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack service show swift -c id -f value"
+      register: service_swift_uuid
+
+- name: Configure object store to use rgw
+  cifmw.general.ci_script:
+    extra_args:
+      KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    output_dir: "/home/zuul/ci-framework-data/artifacts"
+    script: |-
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --project admin --user admin swiftoperator
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user {{ user_swift_uuid.stdout }} --project {{ project_service_uuid.stdout }} member
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user {{ user_swift_uuid.stdout }} --project {{ project_service_uuid.stdout }} admin
+      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack endpoint create --region regionOne {{ service_swift_uuid.stdout }} public http://{{ cifmw_cephadm_vip }}:8080/swift/v1/AUTH_%\(tenant_id\)s
```

Comment:
any update?

Reply:
@fmount latest code configures it use `https` based on the cert/key  availability, by default it will be `http`

see: https://github.com/openstack-k8s-operators/ci-framework/pull/1821/files#diff-48e795627a589458ace8f5d0fdbfad1827cc87421722998b61e157e81e9e8ea4R96